### DISCUSSION TITLE
PyPDF2 2.0.0 development started

### DISCUSSION THREAD

**Discussion by MartinThoma** [py-pdf/pypdf#843] [Announcements]
I'm going to **phase PyPDF2 1.X out**.

That means:

* No new features will be added to PyPDF2 1.X
* PRs that fix bugs will still get merged, but only for May 2022 or until 2.0.0 is out (whatever takes longer)
* Potentially one or more 1.X releases that just introduce deprecation warnings / imports in other places to make the move to PyPDF2 2.0.0 as easy as possible

Meanwhile, the **work on PyPDF2 2.0.0 started**. Changes will get merged into [the `2.0.0-dev` branch](https://github.com/py-pdf/PyPDF2/tree/2.0.0-dev). Once the release is finished, that branch will be merged into the `main` branch.

If you have little experience with git: Don't worry, I'll support you. However, I also hope that this will get finished within May so that we can work directly on the main branch again :-)

**Comment by MartinThoma** [comment]
Question to everybody (and especially @MasterOdin  @pubpub-zz as you both are super active): When should the 2.0.0 be released? Do you think there should be specific features being finished?

Initially, I wanted to have some big cool new feature. For example, better text extraction or some new cryptography support.

However, I think I changed my mind on this. We have already one cool new feature: Types.
The 2.0.0 release will only drop 3.5 and older support. We already have quite a couple of commits and I'd rather release more often than wait for the big changes.

What do you think?

**Reply by pubpub-zz** [reply]
2.0.0 release being the new baseline for development, it should be quite quickly released. Types is a major point, extractText improvements could be released too from my point of view. This version will be an alpha and I hope we will get many feedbacks to increase its reliability.
One major point to make this version apreciated will be to keep API compatibility : a few function (getPage, getObject) seems to be missing.

**Reply by MasterOdin** [reply]
I would say release 2.0.0 once the API with types and all is done and there's no more risk of introducing anything else that's BC breaking. The only thing that need differentiate 2.0.0 between 1.29.0 is some breaking change. I've had major releases of software where the only difference is just dropping an old version of the language and a bunch of bugfixes that benefitted from that and I'll do it again!

**Reply by MartinThoma** [reply]
getPage was not removed on purpose. I added it back: https://github.com/py-pdf/PyPDF2/pull/909

getObject: I need to check. Which one do you mean? (was probably also not on purpose :see_no_evil: )

**Reply by pubpub-zz** [reply]
getObject 'opens' IndirectObject ; quite often it is called implicitely but I've had to call call it sometimes

**Reply by MartinThoma** [reply]
PdfReader.getObject and PdfWriter.getObject still exists. I missed that I removed it accidentially from PdfObject. I'm adding it back: https://github.com/py-pdf/PyPDF2/pull/911

**Reply by MasterOdin** [reply]
One thing I'm confused about is that all of the `PendingDeprecationWarning` say the methods will be removed in `2.0.0`. It seems like that's not the case anymore, and so should that version be changed to `3.0.0` instead?

**Reply by MartinThoma** [reply]
Ah, yes 👍

**Reply by MartinThoma** [reply]
I've adjusted the messages: https://github.com/py-pdf/PyPDF2/pull/919

As this was quite a bit of work (and I want people to move), I'll not adjust the messages for the 1.x branch.

**Comment by MartinThoma** [comment]
The plan is to release 2.0.0 on 1st of June. There was another bugfix for 1.x today. Aparently some people are using the xmp stuff
